In [2]:
library(tidyverse)
library(repr)
source("cleanup.R")
options(repr.matrix.max.rows = 6)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors
Warning message in file(filename, "r", encoding = encoding):
“cannot open file 'cleanup.R': No such file or directory”


ERROR: Error in file(filename, "r", encoding = encoding): cannot open the connection


In [39]:
players <- read_csv("players.csv") |>
select(-hashedEmail, -name) |>
filter(!is.na(Age)) |>
players

summary(players$Age)
summary(players$played_hours)

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


experience subscribe played_hours gender            Age
1   Pro         TRUE     30.3         Male               9 
2   Veteran     TRUE      3.8         Male              17 
3   Veteran    FALSE      0.0         Male              17 
4   Amateur     TRUE      0.7         Female            21 
5   Regular     TRUE      0.1         Male              21 
6   Amateur     TRUE      0.0         Female            17 
7   Regular     TRUE      0.0         Female            19 
8   Amateur    FALSE      0.0         Male              21 
9   Amateur     TRUE      0.1         Male              47 
10  Veteran     TRUE      0.0         Female            22 
11  Veteran     TRUE      1.6         Female            23 
12  Pro         TRUE      0.0         Male              17 
13  Beginner    TRUE      1.5         Female            25 
14  Amateur     TRUE      0.2         Male              22 
15  Veteran     TRUE      0.0         Non-binary        17 
16  Amateur    FALSE      0.0         Male              22 
17  Beginner    TRUE      0.0         Female            17 
18  Amateur     TRUE     48.4         Female            17 
19  Amateur     TRUE      0.5         Male              17 
20  Regular     TRUE      0.6         Male              19 
21  Regular     TRUE      0.3         Male              58 
22  Amateur    FALSE      0.1         Female            17 
23  Beginner    TRUE      1.0         Male              17 
24  Beginner    TRUE      0.0         Male              17 
25  Amateur     TRUE      0.7         Female            21 
26  Regular     TRUE      0.6         Female            28 
27  Regular     TRUE      0.0         Male              17 
28  Veteran    FALSE      0.0         Male              23 
29  Amateur     TRUE      1.8         Male              23 
30  Veteran    FALSE      0.1         Male              18 
⋮   ⋮          ⋮         ⋮            ⋮                 ⋮  
165 Regular     TRUE      0.1         Male              21 
166 Amateur    FALSE      1.0         Male              26 
167 Beginner   FALSE      0.3         Female            17 
168 Veteran     TRUE      0.1         Male              17 
169 Veteran    FALSE      0.1         Male              17 
170 Pro        FALSE      0.4         Male              21 
171 Beginner   FALSE      1.8         Male              32 
172 Veteran     TRUE      0.0         Agender           20 
173 Regular    FALSE      0.0         Male              50 
174 Veteran     TRUE      0.0         Male              17 
175 Amateur     TRUE      0.3         Male              17 
176 Pro        FALSE      0.0         Non-binary        17 
177 Veteran    FALSE      2.7         Non-binary        21 
178 Amateur     TRUE      0.4         Two-Spirited      17 
179 Veteran     TRUE      0.1         Non-binary        17 
180 Amateur    FALSE      2.1         Male              24 
181 Amateur     TRUE      0.8         Female            22 
182 Pro        FALSE      0.2         Male              17 
183 Amateur     TRUE     32.0         Male              22 
184 Pro        FALSE      1.7         Male              17 
185 Regular    FALSE      0.1         Male              18 
186 Veteran     TRUE      0.1         Female            44 
187 Amateur     TRUE      0.0         Male              17 
188 Beginner    TRUE      0.0         Female            17 
189 Amateur    FALSE      0.0         Male              17 
190 Amateur     TRUE      0.0         Male              20 
191 Amateur     TRUE      0.0         Female            17 
192 Veteran    FALSE      0.3         Male              22 
193 Amateur    FALSE      0.0         Prefer not to say 57 
194 Amateur    FALSE      2.3         Male              17

   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
   9.00   17.00   19.00   21.14   22.75   58.00 

   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  0.000   0.000   0.100   5.905   0.600 223.100 

# 1. Data Description:
*note that the summary statistics will be included at the bottom as well as in the written report*

The dataset I will be using for my report is the players.csv dataset as I am going to be answering questions about how we can predict what kinds of players are most likely to contribute more data. In the original players.csv dataset, there are 196 rows and 7 columns, however after tidying the data there are 194 rows and 5 columns. The mean age is 21.14 years old with the minimum being 9 years old and the maximum being 58 years old. The mean hours played is 5.91 hours with the lowest amount of hours being 0 and the most amount of hours being 223.10 hours. In the origial dataset, there are 7 variables. 5 categorical variables which are experience, hashed email, name, gender, and if they are subscribed to the newsletter (subscribe). The other 2 variables are numerical which are played hours and age. The experience variable describes each persons perceived experience with the game, if they have a lot of hours they are likely going to consider themselves more of a "Pro". This creates an issue because perceived skill might not directly relate to how many hours they have played on the game. The subscribe variable is whether or not the player is subscribed to the game related newsletter. This can indicate how interested a person is in Minecraft and could relate to their played hours. The played hours is how many hours the player has spent playing Minecraft, a high amount of hours played will likely reflect a "dedicated" player. The gender variable is what gender each player identifies as. The age variable is how old the player is. 


| Variable | Mean | Minumum | Maximum | Median |
| -------  | ---- | ------- | ------- | ------ | 
| Age (Years)      | 21.14 | 9.00 | 58.00 | 19.00 |
| Played Hours (Hours) | 5.91 | 0 | 223.10 | 0.100 |
|

# 2. Questions

The broad question I will be addressing is **"What "kind" of players are most likely to contribute a large amount of data so that we can target those players in our recruiting efforts"**. This means that I will be trying to find out what variables can predict the amount of played hours.  

The specific question that I have formulated is **"Can the characteristics of players such as expereince, age and gender predict the total amount fo time spent playing Minecraft in the players dataset?"**

By looking at how the predictor variables relate to the amount of hours played per person, we can determine if there are any sort of patterns which can help us predict a players total time playing minecraft based on their characteristics. 

Before using any sort of predictive model, the players.csv dataset needs to be modified so that it is suitable to go through any model. I have already removed variables that I deemed not relevant to the question. However, the categorical variables need to be converted into factors so that they can be put into a predictive model as well. 

# 3. Exploratory Data Analysis & Visualization
*note that the dataset loading, dataset wrangling and mean calculation were all performed above so this section will only be performing exploratory visualizations*

To get a better understanding of the data, 